src/components/labelPrinting/LabelPrintingForm.vue

```diff
@@ -0,0 +1,194 @@
+      >
+        <b-form-input
+          id="number_of_labels"
+          v-model="form.selectedNumberOfLabels"
+          type="number"
+          :min="1"
+          :max="9"
+          placeholder="Please enter a number"
+          required
+        ></b-form-input>
+      </b-form-group>
+
+      <b-form-group
+        id="printer_choice_group"
+        label="Choice of Printer:"
+        label-for="printer_choice"
+        description="Please select which printer you wish to use to print the labels."
+      >
+        <b-form-select
+          id="printer_choice"
+          v-model="form.selectedPrinterId"
+          :options="printerOptions"
+          placeholder="Please select a printer"
+          required
+        ></b-form-select>
+      </b-form-group>
+
+      <b-form-group
+        id="copies_group"
+        label="Number of copies per label:"
+        label-for="copies"
+        description="Please select how many copies of each label you would like printing."
+      >
+        <b-form-input
+          id="copies"
+          v-model="form.copies"
+          type="number"
+          :min="1"
+          :max="10"
+          placeholder="Please select a number"
+          required
+        ></b-form-input>
+      </b-form-group>
+
+      <b-button id="resetButton" type="reset" variant="danger" class="float-left">Reset</b-button>
+
+      <labelPrintingModal
+        id="labelPrintingModal"
+        ref="labelPrintingModal"
+        class="float-right"
+        :disabled="!formValid"
+        v-bind="propsToPass()"
+      ></labelPrintingModal>
+    </b-form>
+  </div>
+</template>
+
+<script>
+import LabelPrintingModal from '@/components/labelPrinting/LabelPrintingModal'
+const MESSAGE_SELECT = 'Please select a option'
+
+export default {
+  name: 'LabelPrintingForm',
+  components: {
+    LabelPrintingModal,
+  },
+  data() {
+    return {
+      form: {
+        barcode: null,
+        selectedSuffixId: null,
+        selectedNumberOfLabels: null,
+        selectedPrinterId: null,
+        copies: null,
+      },
+      suffixOptions: [],
+      printerOptions: [],
+      show: true,
+    }
+  },
+  computed: {
+    formValid() {
+      return !!(
+        this.form.barcode &&
+        this.form.selectedSuffixId &&
+        this.form.selectedNumberOfLabels &&
+        this.form.selectedPrinterId &&
+        this.form.copies
+      )
+    },
+  },
+  created() {
+    this.setSuffixOptions()
+    this.setPrinterNames()
+  },
+  methods: {
+    setSuffixOptions() {
+      let suffixOptions = this.$store.getters.suffixes.map((obj, index) => ({
+        value: index + 1,
+        text: obj['one_character_name'] + ' (' + obj['potential_label_name'] + ')',
+        char: obj['one_character_name'],
+      }))
+      suffixOptions.unshift({ value: null, text: MESSAGE_SELECT })
+      this.suffixOptions = suffixOptions
+    },
```

Comment:
I think this can be greatly simplified:

- There isn't really any reason to go via the store here. The data is entirely static, so you can simply `import SuffixList from '@/config/SuffixList'` within the script tags and use it directly.
- I don't think you need to worry about setting an id, and looking up the value. You can just set the value of the option to be the character itself and use it directly. (Obviously rename the variables)
- Also, as this doesn't change, it may be better set as a prop, rather than as data.

Reply:
Tried to find the documentation of using a function to construct default values for a prop, but its not readily apparent on the props section of the vue docs. Instead it seems confined to a comment in the validation section:
https://vuejs.org/guide/components/props.html#prop-validation

Reply:
I think props get passed in from the parent component, so not sure props would be best fit? But I will simplify it and leave as data for now

Reply:
I have got rid of the id, it just means there can't be a placeholder 'select a suffix', as this needs the value of null